narwhals/_pandas_like/dataframe.py

```diff
@@ -728,3 +728,19 @@ def sample(
                 n=n, frac=fraction, replace=with_replacement, random_state=seed
             )
         )
+
+    def unpivot(
+        self: Self,
+        on: str | list[str] | None,
+        index: str | list[str] | None,
+        variable_name: str | None,
+        value_name: str | None,
+    ) -> Self:
+        return self._from_native_frame(
+            self._native_frame.melt(
+                id_vars=index,
+                value_vars=on,
+                var_name=variable_name if variable_name is not None else "variable",
+                value_name=value_name if value_name is not None else "value",
```

Comment:
That's because passing `None` as default from narwhals DataFrame will overwrite the pandas default value of `"value"`